go/mysql/register-replica.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2022 The Vitess Authors.
```

Comment:
Dolthub? Or is this actually forked from vitess main?

Reply:
This was ported over directly from Vitess main